### PR TITLE
DRAFT: Vectorize & Optimize Coordinate Conversion Functions

### PR DESCRIPTION
**Pull request by philipc2**
Closes #XXX

## Expected Usage

```Python
import uxarray as ux

grid_path = "/path/to/grid.nc"
data_path = "/path/to/data.nc"

uxds = ux.open_dataset(grid_path, data_path)

# this is how you use this function
some_output = uxds.some_function()

# this is another way to use this function
other_output = uxds.some_function(some_param = True)

```

## PR Checklist

**General**
- [ ] An issue is linked created and linked
- [ ] Add appropriate labels
- [ ] Filled out Overview and Expected Usage (if applicable) sections

**Testing**
- [ ] Adequate tests are created if there is new functionality
- [ ] Tests cover all possible logical paths in your function
- [ ] Tests are not too basic (such as simply calling a function and nothing else)

**Documentation**
- [ ] Docstrings have been added to all new functions
- [ ] Docstrings have updated with any function changes
- [ ] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [ ] User functions have been added to `docs/user_api/index.rst`

**Examples**
- [ ] Any new notebook examples added to `docs/examples/` folder
- [ ] Clear the output of all cells before committing
- [ ] New notebook files added to `docs/examples.rst` toctree
- [ ] New notebook files added to new entry in `docs/gallery.yml` with appropriate thumbnail photo in `docs/_static/thumbnails/`

### CODE REVIEW

File: uxarray/grid/coordinates.py [hunk 1]
```diff
@@ -83,40 +88,75 @@ def node_xyz_to_lonlat_rad(node_coord):
     return [d_lon_rad, d_lat_rad]
 
 
-@njit(cache=ENABLE_JIT_CACHE)
-def normalize_in_place(node):
```
Comment (rajeeja) [inline uxarray/grid/coordinates.py:87]: There is another issue #346 where the requirement is `node` be a numpy array. It'd be good to tackle both together.  It has to be done carefully as it breaks numba compatibility.

File: uxarray/grid/coordinates.py [hunk 2]
```diff
@@ -83,40 +88,75 @@ def node_xyz_to_lonlat_rad(node_coord):
...
+#     """
+#     if len(node) != 3:
+#         raise RuntimeError("Input array should have a length of 3: [x, y, z]")
+#     return list(np.array(node) / np.linalg.norm(np.array(node), ord=2))
+
+
+#@njit(cache=ENABLE_JIT_CACHE)
+def normalize_in_place(coords):
+
+    coords_norm = coords / np.linalg.norm(coords, ord=2, axis=1)
```
Comment (rajeeja) [inline uxarray/grid/coordinates.py:147]: I have started a branch with similar WIP code - [here](https://github.com/UXARRAY/uxarray/commit/e4226b902065b5cd33f406ec19bd98719d4aaa69#diff-cdc8fc5f4b48af740c90192c6c361e0fe033e6a7157ff6ee67de9e19b6a99a7dR114)

Comment (philipc2) [reply uxarray/grid/coordinates.py:147]: I'll take a look at this! This branch right here is just a few ideas tossed together for now :)

**Comment by philipc2** [comment]
Closing this PR as it's become stale, will revisit the implementation in a new PR

**Pull request closed without merging**